lib/chat/chat_messenger.dart

```diff
@@ -37,24 +37,87 @@ class ChatMessengerState extends State<ChatMessenger> {
 
   Store<AppState> store;
 
+  List<DocumentSnapshot> _messagesSnapshots;
+  bool _isLoading = false;
+  Message lastMessage;
   String recipient;
 
   File imageFile;
   String imageUrl;
 
   final TextEditingController textInputController = TextEditingController();
+  final ScrollController scrollController = ScrollController();
   final ImagePicker imagePicker = ImagePicker();
 
+  @override
+  void setState(fn) {
+    if (mounted) {
+      super.setState(fn);
```

Comment:
What function is `setState` expecting?

Reply:
This is just anything that is being called setState. This is just being done to prevent setState from ever being called while the chat_messenger is unmounted (which was happening during some of the setState calls of the scrollListener)